theme/theme.inc

```diff
@@ -10,14 +10,17 @@
  */
 function islandora_pdfjs_preprocess_islandora_pdfjs(array &$variables) {
   $fedora_object = $variables['fedora_object'];
+  $dsid = (isset($variables['dsid']) ? $variables['dsid'] : NULL);
 
-  if (isset($fedora_object['OBJ']) && $fedora_object['OBJ']->mimetype == 'application/pdf') {
-    $dsid = 'OBJ';
+  if ($dsid) {
+    $file_url = file_create_url("islandora/object/{$fedora_object->id}/datastream/$dsid/view");
```

Comment:
This one should still go through `url()`, since it goes through the Drupal menu router.